turbo/snapshotsync/freezeblocks/block_snapshots.go

```diff
@@ -188,7 +194,12 @@ func (s *Segment) reopenIdx(dir string) (err error) {
 			return fmt.Errorf("%w, fileName: %s", err, fileName)
 		}
 
-		s.indexes = append(s.indexes, index)
+		if s.Decompressor.ModTime().After(index.ModTime()) {
```

Comment:
let's move this logic to another PR. Because i introduced: 
```
var IncompatibleErr = errors.New("incompatible. can re-build such files by command 'erigon snapshots index'")
```
this error is more useful and gives for user "action to solve the problem". 

Also I added to `erigon snapshots index`: 
```
	if err := freezeblocks.RemoveIncompatibleIndices(dirs); err != nil {
		return err
	}
```
so, it does detect, remove and re-create incompatible/broken/old indices. 

Also maybe `ErrIndexModifiedBeforeSegment` must include - advise for user like: 
```
if you restored files from backup - probably you lost `modtime` attribute. you can configure your backup tool to preserve this attribute - for example: ...
```

Also we don't have this feature now (maybe it's not so useful?)

So, anyway - let's move "modtime" check to another PR.

Reply:
ok

Reply:
Removed here: https://github.com/ledgerwatch/erigon/pull/9456/commits/570184e8f1ef12b6d89cf3d89129e98e3ca049c0

It does feel like a retrograde step though - I think there should be some basic integrity check in the open operation rather than just relying on the higher level code 'knowing what its doing'